cargo-near/src/util/mod.rs

```diff
@@ -276,9 +276,27 @@ pub(crate) fn extract_abi_entries(
     unsafe {
         let lib = libloading::Library::new(dylib_path)?;
         for symbol in near_abi_symbols {
-            let entry: libloading::Symbol<fn() -> near_abi::__private::ChunkedAbiEntry> =
+            let entry: libloading::Symbol<extern "C" fn() -> *const std::ffi::c_char> =
                 lib.get(symbol.as_bytes())?;
-            entries.push(entry().clone());
+            match serde_json::from_str(std::ffi::CStr::from_ptr(entry()).to_str()?) {
+                Ok(entry) => entries.push(entry),
+                Err(err) => {
+                    // unfortunately, we're unable to extract the raw error without Display-ing it first
+                    let mut err_str = err.to_string();
+                    if let Some((msg, rest)) = err_str.rsplit_once(" at line ") {
+                        if let Some((line, col)) = rest.rsplit_once(" column ") {
+                            if line.chars().all(|c| c.is_numeric())
+                                && col.chars().all(|c| c.is_numeric())
+                            {
+                                err_str.truncate(msg.len());
+                                err_str.shrink_to_fit();
+                                anyhow::bail!(err_str);
+                            }
+                        }
+                    }
```

Comment:
Good call on truncating the "at line ... column ..." stuff. It looks weird when there is a schema version mismatch, and all other errors should not happen  Even if they do, line+column do not help much when you can't easily see the intermediary JSON value.

Reply:
Exactly! Weird that there is no way to get the core error from `serde_json::Error`, but this works too.